configs/typescript.js

```diff
@@ -51,15 +51,6 @@ module.exports = {
       },
     ],
 
-    // require explicit boolean checks in conditionals
-    "@typescript-eslint/strict-boolean-expressions": [
-      "error",
-      {
-        allowString: false,
```

Comment:
Why not just set `allowString: true`? You can still easily run into problems with `0` when confusing `number` with `number | undefined` for instance...

Reply:
allowString actually only allows a non-nullable string. there is a separate flag to allow non-nullable strings.

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md

They also allow numbers by default, but not nullable numbers, so depends how we want to handle those. We could set: 

```
  allowString: true,
  allowNumber: true,
  allowNullableObject: true,
  allowNullableBoolean: true,
  allowNullableString: true,
  allowNullableNumber: false,
```